src/test/java/picard/fingerprint/CrosscheckFingerprintsTest.java

```diff
@@ -791,6 +793,162 @@ public void testCheckFiles(final List<File> files, final int expectedRetVal, fin
+        final List<File> vcf_idx_files2 = Arrays.asList(NA12892_and_NA123891_vcf_idx, NA12891_2_vcf_idx, NA12892_1_vcf_idx);
+
+        tests.add(new Object[]{vcf_files1, vcf_idx_files1, 0, 4 * 4, false});
+        tests.add(new Object[]{vcf_files2, vcf_idx_files2, 0, 4 * 4, false});
+
+        // SAM vs VCF
+        final File NA12891_r1_noidx = new File(TEST_DATA_DIR, "index_test/files/NA12891.over.fingerprints.r1.bam");
+        final File NA12891_r1_idx = new File(TEST_DATA_DIR, "index_test/indices/NA12891.over.fingerprints.r1.bam.bai");
+        final File NA12891_r2_noidx = new File(TEST_DATA_DIR, "index_test/files/NA12891.over.fingerprints.r2.bam");
+        final File NA12891_r2_idx = new File(TEST_DATA_DIR, "index_test/indices/NA12891.over.fingerprints.r2.bam.bai");
+        final File NA12892_r1_noidx = new File(TEST_DATA_DIR, "index_test/files/NA12892.over.fingerprints.r1.bam");
+        final File NA12892_r1_idx = new File(TEST_DATA_DIR, "index_test/indices/NA12892.over.fingerprints.r1.bam.bai");
+        final File NA12892_r2_noidx = new File(TEST_DATA_DIR, "index_test/files/NA12892.over.fingerprints.r2.bam");
+        final File NA12892_r2_idx = new File(TEST_DATA_DIR, "index_test/indices/NA12892.over.fingerprints.r2.bam.bai");
+
+        final List<File> sam_vcf_files = Arrays.asList(NA12891_r1_noidx, NA12892_r1_noidx, NA12891_r2_noidx, NA12892_r2_noidx, NA12891_g_vcf_noidx, NA12892_g_vcf_noidx);
+        final List<File> sam_vcf_idx_files = Arrays.asList(NA12891_r1_idx, NA12892_r1_idx, NA12891_r2_idx, NA12892_r2_idx, NA12891_g_vcf_idx, NA12892_g_vcf_idx);
+
+        tests.add(new Object[]{sam_vcf_files, sam_vcf_idx_files, 0, 6 * 6, false});
+
+        // SAM test
+        final List<File> sam_files = Arrays.asList(NA12891_r1_noidx, NA12892_r1_noidx, NA12891_r2_noidx, NA12892_r2_noidx);
+        final List<File> sam_idx_files = Arrays.asList(NA12891_r1_idx, NA12892_r1_idx, NA12891_r2_idx, NA12892_r2_idx);
+        tests.add(new Object[]{sam_files, sam_idx_files, 0, 4*4, true});
+
+        return tests.iterator();
+    }
+
+    /**
+     * Test using an INPUT_INDEX_MAP with explicit indices specified for some/all relevant input files. Tests both VCF and SAM files.
+     */
+    @Test(dataProvider = "explicitIndexPathsInputs")
+    public void testExplicitIndexPaths(final List<File> files, final List<File> indices, final int expectedRetVal,
+                                        final int numberOfSamples, boolean ExpectAllMatch) throws IOException {
+        File metrics = File.createTempFile("Fingerprinting", "test.crosscheck_metrics");
+        metrics.deleteOnExit();
+
+        // Make temp file for INPUT_INDEX_MAP
+        final Path inputIndexMap = File.createTempFile("input_index_map", ".tsv").toPath();
+        IOUtil.deleteOnExit(inputIndexMap);
+
+        // Zip files & indices together
+        List<List<String>> zipped_files = IntStream.range(0, Math.min(files.size(), indices.size()))
```

Comment:
final

and maybe refactor this code for writing index_index map file into its own method, since it's repeated a couple times.

Reply:
Done and done